examples/SearchBox.basic.jsx

```diff
@@ -4,10 +4,11 @@
  *--------------------------------------------------------------------------------------------*/
 import * as React from 'react';
 import { SearchBox, Flex } from '@itwin/itwinui-react';
+import './SearchBox.basic.css';
 
 export default () => {
   return (
-    <Flex style={{ width: '70%' }}>
+    <Flex className='basic-search-box-flex'>
```

Comment:
A large portion of this task is to also remove the extra containers like `Flex` and handle it all in CSS, so that the JSX code is less verbose to read.

Reply:
The same comment also applies to any wrapping `<div>`s. Ideally, those should not be in the JSX too.

Thinking a bit more about this, we might need some kind of convention for "wrapper" styles because it is so common to have those in our demos. This wrapper could even include a default max-width.

Reply:
I noticed while working on this that popover elements don't correctly take styles from the CSS files. This seems to be because they get added to the root DOM and not the shadow DOM. Might be best to just add inline styles to those specific elements, since there aren't that many popover based elements in our examples overall that use non standard styling. Wanted to get your opinion on this though, do you think it would worth the effort to try and add these elements back into the shadow DOM?

Reply:
> do you think it would worth the effort to try and add these elements back into the shadow DOM?

It's at least worth looking into. We explicitly set `<body>` as the portal target in a couple places:

https://github.com/iTwin/iTwinUI/blob/97d9a75dd63449c8122669307209f2732c14925e/examples/index.tsx#L10

https://github.com/iTwin/iTwinUI/blob/97d9a75dd63449c8122669307209f2732c14925e/examples/ColorPicker.advancedPopover.jsx#L66

This was done in order to avoid stacking context issues (see #1480), but it might not be necessary for all popovers.

Reply:
Okay, I'll look into it then.

Reply:
`Flex`s have been removed and replaced, moving on to getting the popover elements to appear in the shadow DOM.

Reply:
If it takes too long, you can leave the inline styles on popovers. We can always look into it in the future.

Reply:
Tested a few different ideas with the portal property for a little bit, and couldn't make the popover portal off of an element in the shadow DOM. Decided to just add the inline styles to the elements for now.

Reply:
Then could you bring back `Flex` in `Popover.focus.jsx`? When inline styles wrap into multiple lines, it really hurts readability

Reply:
Most recent commit brings the `Flex` back.